bindings_generator/src/api.rs

```diff
@@ -119,6 +125,67 @@ impl core::cmp::PartialOrd for Enum {
     }
 }
 
+impl Enum {
+    pub fn strip_common_prefix(&mut self) {
+        let mut variants: Vec<_> = self.values.iter().map(|v| (&v.0[..], *v.1)).collect();
+
+        // Build a map of prefix to occurrence
+        loop {
+            let underscore_index = variants[0].0.chars().enumerate().find_map(|(index, ch)| {
+                if ch == '_' {
+                    Some(index)
+                } else {
+                    None
+                }
+            });
+
+            let underscore_index = match underscore_index {
+                Some(index) if index > 0 => index,
+                Some(_) | None => break,
+            };
+
+            // Get a slice up to and including the `_`
+            let prefix = &variants[0].0[..=underscore_index];
+
+            let common_prefix = if variants.iter().all(|v| v.0.starts_with(prefix)) {
+                prefix
+            } else {
+                break;
+            };
+
+            // remove common prefix from variants
+            variants.iter_mut().for_each(|(ref mut name, _)| {
+                *name = &name[common_prefix.len()..];
+            });
+        }
+
+        let new_variants: HashMap<_, _> = variants
+            .into_iter()
+            .map(|(name, value)| {
+                let starts_with_number = name.chars().next().filter(|c| c.is_numeric()).is_some();
```

Comment:
Maybe it's clearer to write:

```rust
let starts_with_number = name.chars().next().expect("name should not be empty").is_numeric();
```

...? The name would only be empty if *all* variants are the same, or, in other words, there is only one variant. Maybe this should be checked at the beginning of the method, so enums with one single variant are kept intact?

Reply:
Good point, that is the case. updated.